src/picongpu/include/fields/currentDeposition/ZigZag/ZigZag.hpp

```diff
@@ -287,7 +290,8 @@ struct ZigZag
             /* this loop is only needed for 2D, we need a flux in z direction */
             for (uint32_t d = simDim; d < 3; ++d)
             {
-                flux[d] = float_X(0.5) *  charge * velocity[d] * volume_reci;
+                /* in 2D the full flux for the z direction is given to the virtual particle zero */
+                flux[d] = (l == float_X(0) ? charge * velocity[d] * volume_reci : float_X(0.0));
```

Comment:
I don't see why depositing `half` in `flux[d]` should be the same as depositing it fully. that does not look equivalent.